vault/resource_token_auth_backend_role.go

```diff
@@ -0,0 +1,280 @@
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+				Description: "List of disallowed policies for given role.",
+			},
+			"orphan": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Default:     false,
+				Description: "If true, tokens created against this policy will be orphan tokens.",
+			},
+			"period": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				Description: "The duration in which a token should be renewed. At each renewal, the token's TTL will be set to the value of this parameter.",
+			},
+			"renewable": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Default:     true,
+				Description: "Wether to disable the ability of the token to be renewed past its initial TTL.",
+			},
+			"explicit_max_ttl": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				Description: "If set, the token will have an explicit max TTL set upon it.",
+			},
+			"path_suffix": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Default:     "",
+				Description: "Tokens created against this role will have the given suffix as part of their path in addition to the role name.",
+			},
+			"ttl": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				Description: "The TTL period of tokens issued using this role, provided as the number of minutes.",
+			},
+			"max_ttl": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				Description: "The maximum allowed lifetime of tokens issued using this role.",
+			},
+		},
+	}
+}
+
+func tokenAuthBackendRoleCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*api.Client)
+
+	role := d.Get("role_name").(string)
+
+	path := tokenAuthBackendRolePath(role)
+
+	log.Printf("[DEBUG] Writing Token auth backend role %q", path)
+
+	iAllowedPolicies := d.Get("allowed_policies").([]interface{})
+	allowedPolicies := make([]string, 0, len(iAllowedPolicies))
+	for _, iPolicy := range iAllowedPolicies {
+		allowedPolicies = append(allowedPolicies, iPolicy.(string))
+	}
+
+	iDisallowedPolicies := d.Get("disallowed_policies").([]interface{})
+	disallowedPolicies := make([]string, 0, len(iDisallowedPolicies))
+	for _, iPolicy := range iDisallowedPolicies {
+		disallowedPolicies = append(disallowedPolicies, iPolicy.(string))
+	}
+
+	data := map[string]interface{}{}
+
+	if len(allowedPolicies) > 0 {
+		data["allowed_policies"] = allowedPolicies
+	}
+
+	if len(disallowedPolicies) > 0 {
+		data["disallowed_policies"] = disallowedPolicies
+	}
+
+	data["explicit_max_ttl"] = d.Get("explicit_max_ttl").(string)
+	data["ttl"] = d.Get("ttl").(string)
+	data["max_ttl"] = d.Get("max_ttl").(string)
+	data["orphan"] = d.Get("orphan").(bool)
+	data["renewable"] = d.Get("renewable").(bool)
+	data["path_suffix"] = d.Get("path_suffix").(string)
+
+	_, err := client.Logical().Write(path, data)
+
+	d.SetId(path)
```

Comment:
Would it be possible to move this up above the `client.Logical().Write` call so the err below gets checked _right_ after the call? Or move it into an if/else for the err?

Reply:
@tyrannosaurus-becks Done